core/templates/base-components/oppia-footer.component.ts

```diff
@@ -42,12 +51,47 @@ export class OppiaFooterComponent {
     this.router.url === '/about' && !AppConstants.DEV_MODE);
 
   constructor(
+    private alertsService: AlertsService,
+    private ngbModal: NgbModal,
+    private mailingListBackendApiService: MailingListBackendApiService,
+    private platformFeatureService: PlatformFeatureService,
     private router: Router
   ) {}
 
   getOppiaBlogUrl(): string {
     return '/blog';
   }
+
+  validateEmailAddress(): boolean {
+    let regex = new RegExp(AppConstants.EMAIL_REGEX);
+    return regex.test(String(this.emailAddress));
+  }
+
+  subscribeToMailingList(): void {
+    this.mailingListBackendApiService.subscribeUserToMailingList(
+      String(this.emailAddress),
+      // Most of the time, the name is null since it is not provided.
+      String(this.name),
```

Comment:
If you convert null to string, lots of users would be stored as "null"? And I see in the backend that you are using nulls in the APIs. I'd say use nulls everywhere in case of null or empty string to be consistent and ensure that the backend checks for this too.

Reply:
done.